### ISSUE TITLE
str_to_string does not trigger inside a map function

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#16511]
### Summary

If you try to use `ToString::to_string` in a map function on an option, the lint does not trigger.

### Lint Name

`str_to_string`

### Reproducer

I tried this code:

```rust
pub fn m1(x: Option<&str>) -> Option<String> {
    x.map(ToString::to_string)
}
```

https://godbolt.org/z/6f8zz67ac

I expected to see this happen:

```
warning: `to_string()` called on a `&str`
```

Instead, this happened:

Lint did not trigger

### Version

```text

```

**Comment by nyurik** [comment]
fixed in #16512
